pkg/workflows/management/upgrade_management_components_test.go

```diff
@@ -109,12 +112,22 @@ func TestRunnerHappyPath(t *testing.T) {
 		mocks.provider.EXPECT().Name(),
 		mocks.provider.EXPECT().SetupAndValidateUpgradeCluster(ctx, gomock.Any(), newSpec, curSpec),
 		mocks.provider.EXPECT().PreCoreComponentsUpgrade(gomock.Any(), gomock.Any(), gomock.Any()),
-		mocks.capiManager.EXPECT().Upgrade(ctx, managementCluster, mocks.provider, curSpec, newSpec).Return(capiChangeDiff, nil),
-		mocks.gitOpsManager.EXPECT().Install(ctx, managementCluster, curSpec, newSpec).Return(nil),
-		mocks.gitOpsManager.EXPECT().Upgrade(ctx, managementCluster, curSpec, newSpec).Return(fluxChangeDiff, nil),
-		mocks.clusterManager.EXPECT().Upgrade(ctx, managementCluster, curSpec, newSpec).Return(eksaChangeDiff, nil),
-		mocks.eksdUpgrader.EXPECT().Upgrade(ctx, managementCluster, curSpec, newSpec).Return(eksdChangeDiff, nil),
 		mocks.clientFactory.EXPECT().BuildClientFromKubeconfig(managementCluster.KubeconfigFile).Return(mocks.client, nil),
+		mocks.client.EXPECT().Get(ctx, curSpec.EKSARelease.Name, curSpec.EKSARelease.Namespace, &releasev1alpha1.EKSARelease{}).
+			DoAndReturn(func(_ context.Context, _ string, _ string, obj *releasev1alpha1.EKSARelease) error {
+				curSpec.EKSARelease.DeepCopyInto(obj)
+				return nil
+			}),
```

Comment:
why do we need to do this? can't we use a fake client?

Reply:
The test had already been using the fake client, so I continued with using it here. But, yes, the fake client can be used instead.